README.tmpl.md

```diff
@@ -0,0 +1,206 @@
+- `-since` — a git ref specifying that only changes since then should be considered for misspelling (requires git).
+- `-update-dict` — whether the `-misspellings` flag is being used to update a dictionary that already exists.
+- `-write-config` — emit a config file based on flags and existing config to stdout and exit.
+
+
+## Configuration Files
+
+`gospel` uses two configuration file types, `.words` files at the module
+roots of packages that are being checked, and a `.gospel.conf` file at the
+module root of the module in which `gospel` was invoked.
+
+If `gospel` is being used in CI, these files should be committed to the
+repository.
+
+
+### `.words`
+
+The `.words` file is a hunspell `.dic` formatted file. At its simplest, this
+is a plain text file with a list of words, one word per line and an initial
+word count on the first line. The word count is a hint only and does not need
+to match the number of words in the dictionary (a value of `1` will work fine
+if you are constructing the file by hand).
+
+`gospel` will read _all_ `.words` files found at module roots in for the
+packages that are being checked and build a dictionary from them to give to
+hunspell for the spelling checks.
+
+Hunspell dictionaries are able to express more than just word matches though,
+and are able to indicate some grammatically related sets of words based on
+rules. This is covered lightly [below](#hunspell-dictionaries).
+
+
+### `.gospel.conf`
+
+Runtime behaviour of `gospel` can be modified in a persistent way through the
+TOML format `.gospel.conf` file. A number of options are provide:
+
+- `ignore_idents` — whether to include syntax information from the source code in the dictionary of acceptable words.
+- `lang` — the language tag to specify language locale.
+- `show` — whether to show context for identified misspellings.
+- `check_strings` — whether to check string literals.
+- `check_embedded` — whether to check spelling in files embedded using `//go:embed`.
+- `ignore_upper` — whether to ignore words that are all uppercase or their plurals.
+- `ignore_single` — whether to ignore single rune words.
+- `ignore_numbers` — whether to ignore number literals.
+- `mask_flags` — whether words that could be command-line flags should be removed prior to checking.
+- `mask_urls` — whether URLs should be removed prior to checking.
+- `check_urls` — whether the HTTP/HTTPS reachability of URLs should be checked.
+- `camel` — whether to split camelCase words into the components if the complete word is not accepted, otherwise split only on underscore.
+- `max_word_len` — the maximum length of words that should be checked.
+- `min_naked_hex` — minimum length for exclusion of words that are composed of only hex digits 0-9 and a-f (case insensitive).
+- `suggest` — when suggestions should be presented for misspellings: 0 - never, 1 - only the first time the error is found, 2 - always.
```

Comment:
what about using enum-like strings rather than integers, so they are self-describing? `never`, `first`, `always`

Reply:
Yes, I have been thinking about this for a while, but didn't because of breaking. Better to do that earlier than later, I guess.

Reply:
I'll do this in a follow-up.